### PR TITLE
fix: route edit page suspend due to the form shouldupdate

### LINKED ISSUE #1922: special route cause the page suspend
## How to Reproduce

I created a 301 route with data
```
{
  "uris": ["/"],
  "name": "root_path",
  "plugins": {
    "redirect": {
      "ret_code": 302,
      "uri": "/login"
    }
  },
  "labels": {
    "tag": "xxx"
  },
  "status": 1
}
```

I clicked the edit button of this route from the route list,the page appeared suspended animation. After a minute, this window Non-stop pop-ups.
```
word in picture:
Please check the service bound to the route. 
If you do not bind the service, you must set up the upstream service (step 2)
```

image

## Environment

all in docker
* apisix version :  apache/apisix:2.5-alpine
* OS: Ubuntu 18.04
* etcd version: bitnami/etcd:3.4.15
* apisix-dashboard version: apache/apisix-dashboard:2.6
* Browser version: 90.0.4430.93 (x86_64)

## Additional context
when choose the redirect,  can not choose the upsteam and plugins.

### PR DESCRIPTION
**Pull request by liuxiran**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [X] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

1. route with redirect plugin can be create/edit without service and upstream 
2. when notice the route without service and upstream, it should not update the form service_id field

**Related issues**

fix/resolve #1922 

**Checklist:**

- [X] Did you explain what problem does this PR solve? Or what new features have been added?
- [X] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Review by LiteSun — APPROVED** [review]

File: web/src/pages/Route/components/Step1/MetaView.tsx [hunk 1]
```diff
@@ -205,6 +205,7 @@ const MetaView: React.FC<RouteModule.Step1PassProps> = ({ disabled, form, isEdit
             >
               <Select
                 disabled={disabled}
+                data-cy='route-redirect'
                 onChange={(parmas) => {
```
Comment (iamayushdas) [inline web/src/pages/Route/components/Step1/MetaView.tsx:209]: can you also fix the typo to params in this PR?

**Review by iamayushdas — APPROVED** [review]

File: web/src/pages/Route/components/Step1/MetaView.tsx (hunk 1 above)
Comment (liuxiran) [reply web/src/pages/Route/components/Step1/MetaView.tsx:209]: > can you also fix the typo to params in this PR?

Thanks for your review, and just to make sure that I should change the typo of `route-direct` right?

Comment (iamayushdas) [reply web/src/pages/Route/components/Step1/MetaView.tsx:209]: `onChange={(parmas) => {`  see here it should be params i guess

Comment (liuxiran) [reply web/src/pages/Route/components/Step1/MetaView.tsx]: got it, fixed.

Comment (iamayushdas) [reply web/src/pages/Route/components/Step1/MetaView.tsx]: thank you

**Review by iamayushdas — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Review by bzp2010 — APPROVED** [review]

**Pull request merged**